sdk/appconfiguration/Azure.Data.AppConfiguration/src/SettingSelector.cs

```diff
@@ -17,58 +19,223 @@ namespace Azure.Data.AppConfiguration
     /// </summary>
     public class SettingSelector
     {
+        private StringBuilder _keys;
+        private StringBuilder _labels;
+        private SettingFields _fields = SettingFields.All;
+
+        private StringBuilder Keys => _keys ??= new StringBuilder();
+        private StringBuilder Labels => _labels ??= new StringBuilder();
+
+        private const char AnyChar = '*';
+        private const char EscapeChar = '\\';
+        private const char SeparatorChar = ',';
+        private const string KeyQueryFilter = "key";
+        private const string LabelQueryFilter = "label";
+        private const string FieldsQueryFilter = "$select";
+
         /// <summary>
         /// A wildcard that matches any key or any label when passed as a filter
         /// to Keys or Labels.
         /// </summary>
         public static readonly string Any = "*";
```

Comment:
If we are handling wildcards internally, do we want to expose this in the public API still?

Reply:
Probably not.